### PR TITLE
feat: add orion plugin to generate js_binary for main.{js,ts}

### PR DESCRIPTION
**Pull request by jbedard**
Change the "library" package entry points to "lib.{js,ts}" to distinguish them from "apps" that have "main.{js,ts}".

Add an orion plugin to generate `js_binary` whenever a "src/main.{js,ts}".

---

### Test plan

- Covered by existing test cases

### CODE REVIEW

File: pnpm-workspaces/.aspect/cli/main.star [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    entry_point = ctx.sources[0].path.replace(".ts", ".js")
+
+    ctx.targets.add(
+        name = "main",
+        kind = "js_binary",
+        attrs = {
+            # The entry point file
+            "entry_point": entry_point,
+            "data": [
+                aspect.Label(
```
Comment (jbedard) [inline pnpm-workspaces/.aspect/cli/main.star:22]: This should really be:
```
aspect.Import(
   provider = "js",
   id = entry_point
)
```

...then orion should automatically determine where that entry came from and add the correct label. For some reason that isn't working here and I don't have time to look into it atm, but for this demo we know this label will always be correct.

File: pnpm-workspaces/.aspect/cli/main.star [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+                    name = "tsc",
+                ),
+            ],
+        },
+    )
+
+aspect.register_configure_extension(
+    id = "main_js",
+    prepare = lambda _: aspect.PrepareResult(
+        sources = [aspect.SourceFiles("src/main.js", "src/main.ts")],
```
Comment (jbedard) [inline pnpm-workspaces/.aspect/cli/main.star:33]: This could be `SourceGlobs` if you want the paths to be more generic 🤷

Comment (gregmagolan) [reply pnpm-workspaces/.aspect/cli/main.star]: Cool. Updated.

File: pnpm-workspaces/.aspect/cli/main.star (hunk 1 above)
Comment (gregmagolan) [reply pnpm-workspaces/.aspect/cli/main.star:22]: SG

**Review by gregmagolan — APPROVED** [review]
🌮

**Pull request merged**